src/Symfony/Component/Form/FormFactory.php

```diff
@@ -60,9 +60,23 @@ public function createForProperty($class, $property, $data = null, array $option
      */
     public function createBuilder($type = 'form', $data = null, array $options = array())
     {
-        $name = $type instanceof FormTypeInterface || $type instanceof ResolvedFormTypeInterface
-            ? $type->getName()
-            : $type;
+        if (!$type instanceof FormTypeInterface && !$type instanceof ResolvedFormTypeInterface) {
+            $type = $this->registry->getType($type);
+        }
+
+        $typeName = $type->getName();
+
+        if (false === ($pos = strrpos($typeName, '\\'))) {
+            // No FQCN - leave unchanged for BC
+            $name = $typeName;
```

Comment:
are we going to remove that in 3.0? Then we could also remove the `FormType::getName` completely, or?